app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+class ArticlesController < ApplicationController
+  before_action :set_article, only: %i[show edit update destroy]
+
+  def index
+    @articles = Article.sorted.paginate(page: params[:page])
+  end
+
+  def show; end
+
+  def new
+    @article = Article.new
+  end
+
+  def create
+    @article = Article.new(article_params)
+    if @article.save
+      redirect_to @article, notice: I18n.t('controllers.admin.article.create_success')
+    else
+      flash.now[:alert] = I18n.t('controllers.admin.article.create_unsuccess')
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if @article.update(article_params)
+      redirect_to article_path(@article), notice: I18n.t('controllers.admin.article.create_unsuccess')
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    article = set_article
+    if @article.destroy!
+      redirect_to articles_path, notice: "#{I18n.t('controllers.admin.article.destroy_success_1')}'#{article.title}'
```

Comment:
I think it's overdoing, btw, you can pass parameters to i18n so you don't need TWO messages in yml, but only one:
```

I18n.translate('controllers.admin.article.destroy_success', title: article.title)

en:
 destroy_success: "Some text %{title}"
```

Reply:
trust me, current level of i18n is enough, start writing ruby/rails code, and not yaml
there is no profession - i18n Yaml Developer
I see huge gaps in Ruby/Rails and you focusing on yaml